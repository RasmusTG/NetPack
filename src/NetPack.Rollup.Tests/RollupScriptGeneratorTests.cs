using ApprovalTests;
using ApprovalTests.Reporters;
using Dazinator.AspNet.Extensions.FileProviders;
using Microsoft.Extensions.DependencyInjection;
using Newtonsoft.Json.Linq;
using Scriban;
using System.Threading.Tasks;
using Xunit;
using Xunit.Abstractions;

namespace NetPack.Rollup.Tests
{
    [UseReporter(typeof(DiffReporter))]
    public class RollupScriptGeneratorTests
    {

        private readonly ITestOutputHelper _output;


        public RollupScriptGeneratorTests(ITestOutputHelper output)
        {
            _output = output;
        }

        [Fact]
        public async Task Can_Generate_Script()
        {

            var rollupOptions = new RollupPipeOptions();
            dynamic pluginOptions = new Newtonsoft.Json.Linq.JObject();
            pluginOptions.Start = true;
            pluginOptions.Times = 1;
            pluginOptions.Things = new JArray("foo", "bar");
            rollupOptions.AddPlugin("foopackage", pluginOptions);
            rollupOptions.AddPlugin("barpackage", null);

            var path = "./RollupTemplate.txt";
            var templateText = System.IO.File.ReadAllText(path);
            var template = Template.Parse(templateText);
            var result = template.Render(rollupOptions);

            Approvals.Verify(result);

        }     
    }
}