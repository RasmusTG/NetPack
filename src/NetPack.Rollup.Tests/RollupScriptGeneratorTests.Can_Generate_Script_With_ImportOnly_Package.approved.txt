"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const rolluphost_1 = require("netpack-rollup");
const RollupFileOptions_1 = require("netpack-rollup/dist/RollupFileOptions");
const RollupOutputOptions_1 = require("netpack-rollup/dist/RollupOutputOptions");
const hypothetical = require("rollup-plugin-hypothetical");
const fs = require("fs");

const plugin0 = require("foopackage");
const barp = require("barpackage");

var allFiles = {};

module.exports = {

    build: function (callback, requestDto) {

        var files = requestDto.files;
        for (var i = 0; i < requestDto.files.length; i++) {

            var file = requestDto.files[i];
            allFiles[file.path] = file.contents;
        }

        var inputOptionsDto = requestDto.inputOptions;
        var outputOptionsDto = requestDto.outputOptions;

        var inputOptions = new RollupFileOptions_1.default();
        inputOptions.input = inputOptionsDto.input;
        // inputOptions.entry = inputOptionsDto.input;

        var hypotheticalPlugin = hypothetical({
            files: allFiles
        });
        hypotheticalPlugin.cwd = false;
		var plugins = [hypotheticalPlugin];		
				
		var pluginConfig0 = {"Start":true,"Times":1,"Things":["foo","bar"]}
		var newPlugin0 = plugin0(pluginConfig0);		
		plugins.push(newPlugin0);
		

        inputOptions.plugins = plugins;
        var outputOptions = new RollupOutputOptions_1.default();
        outputOptions.format = outputOptionsDto.format;

        let sut = new rolluphost_1.default();
		
        sut.build(inputOptions, outputOptions).then(result => {
            if (result === undefined) {
               callback("result was undefined", null);
            }
            else {
                var code = result.Code;
                var sourceMap = result.SourceMap;
                // response.files.push({ path: property, contents: results[property] })
                var rollupResult = { Code: code, SourceMap: sourceMap };
                callback(null, rollupResult);
            }          

        }).catch((reason) => {
            callback(reason, null);
        });
    }
};